arbnode/inbox_tracker.go

```diff
@@ -102,6 +102,72 @@ func (t *InboxTracker) Initialize() error {
 		return err
 	}
 
+	var dbBatchParsingVersion uint64
+	hasKey, err = t.db.Has(batchParsingVersion)
+	if err != nil {
+		return err
+	}
+	if hasKey {
+		value, err := t.db.Get(batchParsingVersion)
+		if err != nil {
+			return err
+		}
+		err = rlp.DecodeBytes(value, &dbBatchParsingVersion)
+		if err != nil {
+			return err
+		}
+	}
+	if dbBatchParsingVersion < currentBatchParsingVersion {
+		log.Info("Upgrading database batch parsing version", "oldBatchParsingVesrion", dbBatchParsingVersion, "newBatchParsingVersion", currentBatchParsingVersion)
+		err = t.clearUnprocessedBatches()
+		if err != nil {
+			return fmt.Errorf("failed to clear unprocessed batches: %w", err)
+		}
+		value, err := rlp.EncodeToBytes(currentBatchParsingVersion)
+		if err != nil {
+			return err
+		}
+		err = t.db.Put(batchParsingVersion, value)
+		if err != nil {
+			return err
+		}
+	} else if dbBatchParsingVersion > currentBatchParsingVersion {
+		return fmt.Errorf("database has unsupported batch parsing version %v (this node version only supports version %v)", dbBatchParsingVersion, currentBatchParsingVersion)
+	}
+
+	return nil
+}
+
+// clearUnprocessedBatches reorgs to the batch of the latest block
+func (t *InboxTracker) clearUnprocessedBatches() error {
```

Comment:
The name function and doc is a little confusing.
Should either add documentation about it's usage - only on startup, to remove batches suspected of being parsed with the wrong version, and/or change function name to hint at it - I can't think of a good name but maybe just add OnStartUp and have a comment explaining the rationale

Reply:
I've added a detailed comment explaining the rationale. I don't love modifying the name, because the function itself doesn't require it's invoked on startup, even though that's how it's used